server/udns_server.ml

```diff
@@ -466,10 +480,19 @@ module Notification = struct
       | xs -> Domain_name.Map.add name xs new_map)
       conn Domain_name.Map.empty
 
+  let encode_and_sign key_opt server now packet =
+    let buf, max_size = Packet.encode `Tcp packet in
+    match key_opt with
+    | None -> buf, None
+    | Some key ->
+      match sign_outgoing ~max_size server key now packet buf with
+      | None -> buf, None
+      | Some (out, mac) -> out, Some mac
+
   (* outstanding notifications, with timestamp and retry count
      (at most one per zone per ip) *)
   type outstanding =
-    (int64 * int * Packet.t * Domain_name.t option) Domain_name.Map.t IPM.t
+    (int64 * int * Cstruct.t option * Packet.t * Domain_name.t option) Domain_name.Map.t IPM.t
```

Comment:
I assume `int64` is timestamp, `int` is retry count, maybe document what the `Cstruct.option` is?